### PR TITLE
HTML reader: support <bdo>

### PR DESCRIPTION
**Pull request by TristanCacqueray**
See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/bdo

Closes: #5794

### CODE REVIEW

File: test/command/1592.md [hunk 1]
```diff
@@ -47,3 +47,9 @@ pandoc -f native -t markdown
 [hi]{.smallcaps}
 ```
 
+```
+% pandoc -f html -t native
+<bdo dir="ltr">foo</bdo>
+^D
+[Plain [Span ("",[],[("dir","ltr")]) [Str "foo"]]]
```
Comment (TristanCacqueray) [inline test/command/1592.md:54]: Note that I can't explain why this is wrapped in a `[Plain ...]` block.

File: src/Text/Pandoc/Readers/HTML.hs [hunk 2]
```diff
@@ -802,6 +803,14 @@ pCode = try $ do
   result <- manyTill pAny (pCloses open)
   return $ B.codeWith (mkAttr attr) $ T.unwords $ T.lines $ innerText result
 
+pBdo :: PandocMonad m => TagParser m Inlines
+pBdo = try $ do
+  TagOpen _ attr' <- lookAhead $ pSatisfy $ tagOpen (=="bdo") (const True)
+  let attr = toStringAttr attr'
+  contents <- pInTags "bdo" inline
+  let dir = fromMaybe "" $ lookup "dir" attr
+  return $ B.spanWith ("", [], [("dir",dir)]) contents
```
Comment (mb21) [inline src/Text/Pandoc/Readers/HTML.hs:812]: nit-pick... without the `"dir"` attribute, this is uselels, so maybe do a `case` on the `Maybe` returned from `lookup` ?

Comment (TristanCacqueray) [reply src/Text/Pandoc/Readers/HTML.hs:812]: Alright. Then I guess when dir is different than rtl, bdo is also useless. Next commit fixes that.

Comment (mb21) [reply src/Text/Pandoc/Readers/HTML.hs]: well... if the whole doc is rtl, then a ltr span is useful :-)

Comment (TristanCacqueray) [reply src/Text/Pandoc/Readers/HTML.hs]: Ah, good point! Ok let me not skip rtl then :)  What about validating that dir must be either rtl or ltr? should that be done in the `pBdo` ?

Comment (mb21) [reply src/Text/Pandoc/Readers/HTML.hs]: hm.. good question... I guess it wouldn't hurt.. although we're not doing much validation on most attributes...

File: test/command/1592.md (hunk 1 above)
Comment (despresc) [reply test/command/1592.md:54]: The readers all return a full `Pandoc` document, so a bare inline expression has to be wrapped in some kind of block. This ends up being `Plain` for the html reader.

Comment (TristanCacqueray) [reply test/command/1592.md:54]: If I understand correctly I don't need to update the `pBdo` and this output is expected?

Comment (despresc) [reply test/command/1592.md:54]: It's expected, yes. You'd have to modify `readHtml` or `writeNative` themselves not to have it.

**Comment by jgm** [comment]
Great, thanks!

**Comment by TristanCacqueray** [comment]
Thank you for putting the `good first issue` label!

**Pull request merged**